agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -1023,56 +1028,46 @@ impl DeriveSchema for TaggedOperator {
                     })
             }};
         }
-        macro_rules! array_element_schema_or_error {
-            ($input_schema:expr,$input:expr) => {{
-                match $input_schema {
-                    Schema::Array(a) => *a,
-                    Schema::AnyOf(ao) => {
-                        let mut array_type = None;
-                        for schema in ao.iter() {
-                            if let Schema::Array(a) = schema {
-                                array_type = Some(*a.clone());
-                                break;
-                            }
-                        }
-                        match array_type {
-                            Some(t) => t,
-                            None => {
-                                return Err(Error::InvalidExpressionForField(
-                                    format!("{:?}", $input),
-                                    "input",
-                                ))
-                            }
-                        }
+        match self {
+            TaggedOperator::Convert(c) => {
+                let mut convert_type = match c.to.as_ref() {
+                    Expression::Literal(LiteralValue::String(s)) => schema_for_type_str(s.as_str()),
+                    Expression::Literal(LiteralValue::Double(d)) => {
+                        schema_for_type_numeric(*d as i32)
                     }
-                    _ => {
-                        return Err(Error::InvalidExpressionForField(
-                            format!("{:?}", $input),
-                            "input",
-                        ))
+                    Expression::Literal(LiteralValue::Int32(i)) => schema_for_type_numeric(*i),
+                    Expression::Literal(LiteralValue::Int64(i)) => {
+                        schema_for_type_numeric(*i as i32)
                     }
+                    Expression::Literal(LiteralValue::Decimal128(d)) => {
+                        let decimal_string = d.to_string();
+                        let decimal_as_double = decimal_string
+                            .parse::<f64>()
+                            .map_err(|_| Error::InvalidConvertTypeValue(decimal_string))?;
+                        schema_for_type_numeric(decimal_as_double as i32)
+                    }
+                    // unfortunately, convert can take any expression as a to type. So
```

Comment:
So...?

Reply:
LOL. must have cmd+k'd the second line of this comment at some point, thanks